### PR TITLE
Add dialog:merge tests

### LINKED ISSUE #2636: [PORT] Test dialog:merge in integration builds
> Port this change from botbuilder-dotnet/master branch:
https://github.com/microsoft/botbuilder-dotnet/pull/4334

We have checked-in some bad schema even though dialog:merge verifies and there is a test for it.  The test was disabled on the build server, but is now enabled by installing the bf tool.  This check-in also cleans-up the bad schema.

# Changed projects
* Microsoft.Bot.Builder.Dialogs.Adaptive
* Microsoft.Bot.Builder.Dialogs.Declarative.Tests
* Microsoft.Bot.Builder.TestBot.Json
*

### LINKED ISSUE #3405: port: Enforce that tests.schema must be up to date (#5317)
The changes in [Enforce that tests.schema must be up to date (#5317)](https://github.com/microsoft/botbuilder-dotnet/pull/5317) may need to be ported to maintain parity with `microsoft/botbuilder-dotnet`.

<blockquote>
We have tests that merge all schemas and then test dialog files against them.  This only works if people run tests locally before they check-in.  This changes the behavior so the tests will fail on the builder server if the schema changes. This was tested by uploading with a bad schema (which failed) and then uploading the updated schema (which succeeded.)
</blockquote>

Please review and, if necessary, port the changes.

### PR DESCRIPTION
**Pull request by mdrichardson**
Replaces: https://github.com/microsoft/botbuilder-js/pull/3399

Fixes #2636
Fixes #3405 

## Description

Ported over .NET's/BF-CLI's `dialog:merge` tests and added to CI workflow (only on the Node v10.x run).

### CODE REVIEW

File: libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js [hunk 1]
```diff
@@ -56,12 +56,12 @@ describe('Schema Merge Tests', function () {
...
     // This only runs on Windows platforms.
     maybeIt('should generate a new tests.schema file', async function () {
-        this.timeout(100000);
+        this.timeout(150000);
         fs.unlinkSync(testsSchemaPath);
 
         // Merge all schema files.
         const sdkRoot = path.join(__dirname, '..', '..', '..');
-        const mergeCommand = `bf dialog:merge ${sdkRoot}/libraries/**/*.schema ${sdkRoot}/libraries/**/*.uischema !**/testbot.schema !${sdkRoot}/libraries/botbuilder-dialogs-adaptive/tests/schema/*.* -o ${testsSchemaPath}`;
+        let mergeCommand = `bf dialog:merge ${sdkRoot}/libraries/**/*.schema ${sdkRoot}/libraries/**/*.uischema !**/testbot.schema !${sdkRoot}/libraries/botbuilder-dialogs-adaptive/tests/schema/*.* -o ${testsSchemaPath}`;
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js:64]: One final suggestion - I tend to prefer this style for long commands:
```typescript
const mergeCommand = [
  'npx -p @microsoft/botframework-cli', // invoke with npx to not alter repo dependencies
  'bf dialog:merge', // dialog merge command
  ... // other flags/arguments on separate lines
].join(' ');
```

Comment (joshgummersall) [reply libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js:64]: This whole long string ^ should be broken up and placed in the array, and the keyword should be `const` 😄

Comment (mdrichardson) [reply libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js:64]: @joshgummersall [Beat ya](https://github.com/microsoft/botbuilder-js/pull/3404/commits/35e39dc9e9a4b6a5061ba5c329b0d39168628946) 👍

Comment (joshgummersall) [reply libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js:64]: Top speed achieved!

**Comment by mdrichardson** [comment]
Converting to draft while I add in this: https://github.com/microsoft/botbuilder-js/issues/3405

File: libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js (hunk 1 above)
Comment (joshgummersall) [reply libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js:64]: Not sure if I'm missing something in the diff, but my suggestion is to do:
```typescript
const mergeCommand = [
  'npx -p @microsoft/botframework-cli',
  'bf dialog:merge',
  `${sdkRoot}/libraries/**/*.schema`, // include all schema files
  `${sdkRoot}/libraries/**/*.uischema`, // include all uischema files
  '!**/testbot.schema', // ignore testbot schema files
  `!${sdkRoot}/libraries/botbuilder-dialogs-adaptive/tests/schema/*.*`, // ignore teams schema files
  `-o ${testsSchemaPath}` // write output to test schema path
].join(' ');
```

Comment (mdrichardson) [reply libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js]: *Actually* [resolved](https://github.com/microsoft/botbuilder-js/pull/3404/commits/f11c032be39cdf2eac9390c68654908068eab964#diff-99d1dec39793b38d549b790cc1bb6f2bc962dd39aee498126f98b5421456e616R65)

**Review by joshgummersall — APPROVED** [review]

**Pull request merged**